plugins/org.jkiss.dbeaver.ui.editors.data/src/org/jkiss/dbeaver/ui/controls/resultset/ResultSetModel.java

```diff
@@ -276,7 +276,8 @@ public void setAttributeVisibility(@NotNull DBDAttributeBinding attribute, boole
             constraint.setVisible(visible);
             if (attribute.getParentObject() == null || attribute.getParentObject() == documentAttribute) {
                 if (visible) {
-                    visibleAttributes.add(attribute);
+                    final int position = Math.min(constraint.getVisualPosition(), visibleAttributes.size());
```

Comment:
What does constraint.getVisualPosition() return if constraint.isVisible() == false? Will not it turn out that visualPosition is less than zero or something like that?

Reply:
No. It returns the current visual position. You can see it in the filter panel even when the attribute is hidden.